### PR TITLE
CFY-7416 Add tests for the executor module

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

File: tests/test_executor.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    executor.execute(mocked_env, 'workflow_name')
+
+    workflow_runner.WorkflowRunner.assert_called_once_with(
+        'model_storage', 'resource_storage', 'plugin_manager',
+        service_id='service_id',
+        workflow_name='workflow_name',
+        executor=mocker.ANY
+    )
+
+    mock_runner.execute.assert_called_once_with()
```
Comment (aviaefrat) [inline tests/test_executor.py]: no need for `with`

File: tests/test_executor.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        executor.execute(mocked_env, 'workflow_name')
+
+    workflow_runner.WorkflowRunner.assert_called_once_with(
+        'model_storage', 'resource_storage', 'plugin_manager',
+        service_id='service_id',
+        workflow_name='workflow_name',
+        executor=mocker.ANY
+    )
+
+    mock_runner.execute.assert_called_once_with()
```
Comment (aviaefrat) [inline tests/test_executor.py]: with

File: tests/test_executor.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+def test_execution_logging(mocker, mocked_env):
+    mock_runner = _patch_runner(mocker)
+    # Give the process some time to enter and prints logs
```
Comment (aviaefrat) [inline tests/test_executor.py]: print

Comment (aviaefrat) [inline tests/test_executor.py]: Improve this explanation.

File: tests/test_executor.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    mocked_log = mocker.MagicMock()
+    mocked_log.level = 'INFO'
+    mocked_log.traceback = 'traceback'
+    mocker.patch('aria.cli.logger.ModelLogIterator',
+                 return_value=iter([mocked_log]))
+
+    executor.execute(mocked_env, 'workflow_name')
+
+    mocked_env.ctx_logger.info.assert_any_call(mocked_log)
+    mocked_env.ctx_logger.info.assert_called_with('traceback')
```
Comment (aviaefrat) [inline tests/test_executor.py:102]: add call count == 2

**Review by aviaefrat — CHANGES_REQUESTED** [review]

**Review by aviaefrat — APPROVED** [review]

File: tests/test_executor.py (hunk 1 above)
Comment (funkyHat) [reply tests/test_executor.py]: `assert_called_once_with()` is a stronger assertion. Removing it wouldn't be quite the same test.

**Pull request merged**